### PR TITLE
Add websocket to API

### PR DESCRIPTION
**Pull request by haykam821**
This pull request implements the `/api/socket` endpoint, which is a websocket.

- [x] Server → client mirrors `/api/stream`
- [x] Client → server mirrors `/api/message`

### CODE REVIEW

File: bridge/api/api.go [hunk 1]
```diff
@@ -44,6 +45,7 @@ func New(cfg *bridge.Config) bridge.Bridger {
 	e.GET("/api/health", b.handleHealthcheck)
 	e.GET("/api/messages", b.handleMessages)
 	e.GET("/api/stream", b.handleStream)
+	e.GET("/api/socket", b.handleWebsocket)
```
Comment (42wim) [inline bridge/api/api.go]:
```suggestion
	e.GET("/api/websocket", b.handleWebsocket)
```
I think this is a better name

**Comment by 42wim** [comment]
Thanks for the PR, looks good.
Take a look at the linting errors in the Travis CI build though.

File: bridge/api/api.go (hunk 1 above)
Comment (haykam821) [reply bridge/api/api.go]: Renamed in df6875d.

**Comment by haykam821** [comment]
I'll fix the lint errors.

This is still a draft for now since I haven't implemented the server-bound functionality to send messages to Matterbridge.

**Comment by NikkyAI** [comment]
so did the travis-ci crash or what happened ?

**Comment by qaisjp** [comment]
I don't think we use Travis anymore.

**Comment by NikkyAI** [comment]
then maybe it should be changed to be not a `required` check anymore
or is this because this PR is a old draft?

**Comment by 42wim** [comment]
@NikkyAI If you can tell me how to do that let me know .. I removed everything from travis and it's still there.

**Comment by 42wim** [comment]
Still two linting issues, you can fix those by doing `_ = conn.WriteJSON(msg)`
As you implemented it now, it's blocking on the reader, so if you as client don't send any messages, you can also read no new messages from matterbridge.

Maybe https://github.com/gorilla/websocket/blob/master/examples/chat/client.go can be an inspiration?

**Comment by 42wim** [comment]
LGTM, thanks

**Pull request merged**